### ISSUE TITLE
--lock-file or --sync-fd without --unshare-pid results in exit status 1

### ISSUE THREAD

**Issue by smcv** [containers/bubblewrap#336]
If I use `--lock-file` or `--sync-fd` without `--unshare-pid`, I get exit status 1 instead of the exit status of the wrapped command.

```
$ bwrap --ro-bind / / --lock-file /tmp/.ref --unshare-pid echo hello; echo $?
hello
0
$ bwrap --ro-bind / / --lock-file /tmp/.ref echo hello; echo $?              
hello
1
$ bwrap --ro-bind / / --lock-file /tmp/.ref sh -c 'exit 23'; echo $?
1
$ bwrap --ro-bind / / --lock-file /tmp/.ref --unshare-pid sh -c 'exit 23'; echo $?
23
```

Similarly,

```
$ bwrap --ro-bind / / --sync-fd 3 echo hello 3</tmp/.ref; echo $?
hello
1
```

**Comment by smcv** [comment]
Analysis:

`lock_files != NULL` or `opt_sync_fd != -1` take us into the code path where a process calls `do_init()`. In the shared pid namespace case, it isn't really init, but the wrapped process is its direct child, so the wrapped process is reaped by `wait()` anyway.

However, `do_init()` only propagates the exit status from the wrapped process to its own exit status if `event_fd` is open, which is only done in the `--unshare-pid` case.

**Comment by smcv** [comment]
Looks like this is probably fixed in #325.

**Comment by smcv** [comment]
Fixed in 0.4.1
